rdkcentral/Lightning-UI-Components
# fix(TextBox): ensure component adheres to width

## Description


A bug was encountered where just setting w on the `TextBox` component does not actually change the width of the component. Only after applying `wordWrapWidth` does the `TextBox` update its width.

This PR ensures that if a user defines the width (either with `w` or `wordWrapWidth` in `textStyles`) `wordWrapWidth` will be set on the `TextBox` component. This way the text will actually conform to the set width (even when no explicit `wordWrapWidth` is defined).

A `fixed` prop was added to `TextBox` similar to the `fixed` prop in `Button` that will need to be set to `true` to update the text width if the width of the text is set by providing a `w` value.

Updates to the `TextBox` documentation were also done in this PR in accordance to LUI-657.
Likewise a section on the updates to the width/different cases of setting the width were also added to the `TextBox` documentation.

## References


LUI-660 and LUI-657

## Testing


Go to the `TextBox` component

- Ensure initial width is 600
- Ensure fixed is set to true
- In the `TextBox` story, update the initial `w` value to a different value (eg 400) and confirm the width of the text updates to the set width value
- Remove the fixed prop setter and confirm that the text is no longer confirming to the set `w` value but should be following the rendered width
- In the `TextBox` story, remove the `w` value and add `wordWrapWidth` and a value to the `textStyle` object in the Basic story and confirm the width of the component reflects the set `wordWrapWidth` value

For example:
```js
TextBox: {
  type: TextBox,
  style: { textStyle: { wordWrapWidth: 500, maxLines: 3 } },
}
```

- `TextBox`'s `InlineContent`'s w should not be changing/the entire component should be showing even when w is set to a small value (this work/issue will be addressed in another ticket)
- Modify and add spaces to the title in `Button` and there shouldn't be truncation on the text until fixed is set to true (test ex strings: "1234567898786756576788985678987 Button" and "To Be Removed")
- Modify and add spaces to the title in `Tooltip`, the spaces should not make new lines and the text should be continuous on 1 line (eg a title like 'Longer Tooltip' should all be on 1 line and not on 2 lines)

**For all components that utilize `TextBox` (`Button`, `Tooltip`, etc), ensure the text is centered in the middle of the component.

## Automation



## Checklist
